pkg/server/event_handler_test.go

```diff
@@ -468,12 +498,42 @@ func (p *successProcessor) Process(_ context.Context, m *structpb.Struct) error
 }
 
 // failMessenger Send always fail.
-type failMessenger struct{}
 
-func (m *failMessenger) Send(_ context.Context, _ *v1alpha1.PmapEvent) error {
+type testMessanger interface {
+	Messenger
+	getPmapEvent() *v1alpha1.PmapEvent
+}
+
+type failMessenger struct {
+	gotPmapEvent *v1alpha1.PmapEvent
+}
```

Comment:
Can we just have one fake messenger? The only difference between the two is one requires error. You achieve the same with a field. E.g.

```go
type testMessenger struct {
  returnErr error
  gotEvent *v1alpha1.PmapEvent
}

func (m *testMessenger) Send(_ context.Context, p *v1alpha1.PmapEvent) error {
  m.gotEvent = p
  return m.returnErr
}

// We should get rid of these in a separate refactor
// See https://github.com/abcxyz/pmap/issues/96
func (m *testMessenger) Cleanup() error {
	return nil
}
```